lib/ansible/module_utils/facts/packages.py

```diff
@@ -19,7 +19,7 @@ def get_all_pkg_managers():
     return {obj.__name__.lower(): obj for obj in get_all_subclasses(PkgMgr) if obj not in (CLIMgr, LibMgr, RespawningLibMgr)}
 
 
-class PkgMgr(with_metaclass(ABCMeta, object)):  # type: ignore[misc]
+class PkgMgr(metaclass=ABCMeta):
```

Comment:
I learned [recently](https://github.com/cherrypy/cheroot/commit/b0b4754317120ff735605e1127bfcfacf589232d) from [`refurb`](https://docs.astral.sh/ruff/rules/meta-class-abc-meta/) that it's cleaner to inherit from `abc.ABC` instead:
```suggestion
class PkgMgr(ABC):
```

Reply:
Fine with me but I am trying to not change anything outside of plain search/replace of `six` usage unless I really have to, not make this PR even bigger.

Reply:
I agree it's worth keeping these changes to a minimum, as it's already a large set of changes.

Also, using `ABC` over `metaclass=ABCMeta` isn't always preferable. Using `metaclass=ABCMeta` on derived types acts as a hint to some tools that the type is intended to remain abstract, which prevents them from reporting that the type is incomplete. That doesn't apply in this particular case, but I wanted to point it out, since the advise from `refurb` does not take this into account.